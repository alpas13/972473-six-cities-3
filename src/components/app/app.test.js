import React from "react";
import rerender from "react-test-renderer";
import App from "./app";
import Main from "../main/main";
import OffersList from "../offers-list/offers-list";
import Map from "../map/map";

const mock = {
  offers: [
    {
      userName: `user-name 1`,
      propertyImage: [`path 1`],
      title: `Title text 1`,
      mark: `mark 1`,
      previewImage: `path 1`,
      price: 1,
      bookmark: false,
      rating: {
        star: 1,
        value: 1,
      },
      features: {
        entire: `type text 1`,
        bedrooms: 1,
        adults: 1,
      },
      insideList: [
        `text 1`,
      ],
      hostName: `text 1`,
      propertyText: [
        `text 1`,
        `text 1`
      ],
      coords: [
        52.3909553943508,
        4.85309666406198
      ]
    }
  ]
};

test(`Render App correctly`, () => {
  const {offers} = mock;
  const tree = rerender.create(
      <App
        offersCount={3}
        offers={offers}
        city={`Amsterdam`}
      >
        <Main>
          <OffersList
            offersCount={3}
            offers={offers}
            city={`Amsterdam`}
            onTitleOfferClick={() => {}}
          />
          <Map offers={offers} />
        </Main>
      </App>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
