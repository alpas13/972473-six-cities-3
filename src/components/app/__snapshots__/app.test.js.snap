// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App correctly 1`] = `
<div
  className="page page--gray page--main"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="#"
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                  onClick={[Function]}
                >
                  Oliver.conner@gmail.com
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--index"
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <div
      className="tabs"
    >
      <section
        className="locations container"
      >
        <ul
          className="locations__list tabs__list"
        />
      </section>
    </div>
    <div
      className="cities"
    >
      <div
        className="cities__places-container container"
      >
        <section
          className="cities__places places"
        >
          <h2
            className="visually-hidden"
          >
            Places
          </h2>
          <b
            className="places__found"
          >
            0
             places to stay in 
            
          </b>
          <form
            action="#"
            className="places__sorting"
            method="get"
            name="sorting"
            onClick={[Function]}
          >
            <span
              className="places__sorting-caption"
            >
              Sort by
            </span>
            <span
              className="places__sorting-type"
              onClick={[Function]}
              tabIndex="0"
            >
              Popular
              <svg
                className="places__sorting-arrow"
                height="4"
                width="7"
              >
                <use
                  xlinkHref="#icon-arrow-select"
                />
              </svg>
            </span>
            <ul
              className="places__options places__options--custom"
              onClick={[Function]}
            >
              <li
                className="places__option places__option--active"
                data-sort="popular"
                tabIndex="0"
              >
                Popular
              </li>
              <li
                className="places__option"
                data-sort="to-high"
                tabIndex="0"
              >
                Price: low to high
              </li>
              <li
                className="places__option"
                data-sort="to-low"
                tabIndex="0"
              >
                Price: high to low
              </li>
              <li
                className="places__option"
                data-sort="top-rated"
                tabIndex="0"
              >
                Top rated first
              </li>
            </ul>
            <select
              className="places__sorting-type visually-hidden"
              id="places-sorting"
            >
              <option
                className="places__option"
                defaultValue={true}
                value="popular"
              >
                Popular
              </option>
              <option
                className="places__option"
                value="to-high"
              >
                Price: low to high
              </option>
              <option
                className="places__option"
                value="to-low"
              >
                Price: high to low
              </option>
              <option
                className="places__option"
                value="top-rated"
              >
                Top rated first
              </option>
            </select>
          </form>
          <div
            className="cities__places-list places__list tabs__content"
            onMouseLeave={[Function]}
          />
        </section>
        <div
          className="cities__right-section"
        >
          <section
            className="cities__map map"
          >
            <div
              style={
                Object {
                  "height": "800px",
                }
              }
            />
          </section>
        </div>
      </div>
    </div>
  </main>
</div>
`;
